### PR TITLE
[PAG-710] add apim

### PR DESCRIPTION
**Pull request by pasqualespica**
The purpose of this PR is to add APIM according to [Nuova architettura pagoPA](https://pagopa.atlassian.net/wiki/spaces/PAG/pages/384141796/Nuova+architettura+pagoPA)
### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

---

### CODE REVIEW

**Review by pasqualedevita — APPROVED** [review]
lgtm

File: src/env/prod/terraform.tfvars [hunk 1]
```diff
@@ -15,8 +15,17 @@ cidr_subnet_appgateway = ["10.1.128.0/24"]
...
 cidr_subnet_apim      = ["10.230.7.0/26"]
+cidr_subnet_redis     = ["10.1.132.0/24"]
 
 external_domain = "pagopa.it"
 dns_zone_prefix = "platform"
 
 lock_enable = true
+
+apim_publisher_name            = "pagoPA Platform PROD"
+apim_sku                       = "Premium_1"
```
Comment (pasqualedevita) [inline src/env/prod/terraform.tfvars:26]:
```suggestion
apim_sku                       = "Developer_1" # change to Premium_1 before launch
```

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/env/uat/terraform.tfvars [hunk 2]
```diff
@@ -15,6 +15,15 @@ cidr_subnet_appgateway = ["10.1.128.0/24"]
 # https://www.davidc.net/sites/default/subnets/subnets.html?network=10.230.7.0&mask=24&division=7.31
 cidr_vnet_integration = ["10.230.7.0/24"] # ask to SIA
 cidr_subnet_apim      = ["10.230.7.0/26"]
+cidr_subnet_redis     = ["10.1.132.0/24"]
```
Comment (pasqualedevita) [inline src/env/uat/terraform.tfvars:18]: please move under cidr_subnet_appgateway

**Review by pasqualedevita — CHANGES_REQUESTED** [review]

File: src/apim.tf [hunk 3]
```diff
@@ -9,3 +9,176 @@ module "apim_snet" {
...
+  location = var.location
+
+  tags = var.tags
+}
+
+locals {
+  apim_cert_name_proxy_endpoint   = format("%s-proxy-endpoint-cert", local.project)
+  portal_cert_name_proxy_endpoint = format("%s-proxy-endpoint-cert", "portal")
+
+  api_domain        = var.env_short == "p" ? "api.platoform.pagopa.it" : format("api.%s.platoform.pagopa.it", lower(var.tags["Environment"]))
```
Comment (infantesimone) [inline src/apim.tf:24]: typo :-)

**Review by pasqualedevita — APPROVED** [review]
Is this pr applied?
If yes we can merge

**Comment by pasqualespica** [comment]
> Is this pr applied?
> If yes we can merge

applied in `dev`

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**